hibernate-reactive-core/src/test/java/org/hibernate/reactive/testing/ReactiveAssertions.java

```diff
@@ -0,0 +1,102 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.testing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+
+import io.smallrye.mutiny.Uni;
+
+/**
+ * Utility to handle verifying the information about an expected {@link Throwable}
+ */
+public class ReactiveAssertions {
+
+	/**
+	 * Assert that the expected exception type is thrown.
+	 * If the thrown exception is a {@link CompletionException}, then the cause will be tested.
+	 *
+	 * @return a {@link CompletionStage} with the expected exception thrown by the stage we were testing.
+	 */
+	public static <T extends Throwable> CompletionStage<T> assertThrown(Class<T> expectedException, CompletionStage<?> stage) {
+		return stage
+				// Depending on the use case, the real exception could be wrapped in a CompletionException
+				.handle( (v, e) -> e instanceof CompletionException ? e.getCause() : e )
+				.thenApply( e -> {
+					assertThat( e ).isInstanceOf( expectedException );
+					return (T) e;
+				} );
+	}
+
+	/**
+	 * Assert that the expected exception type is thrown.
+	 * If the thrown exception is a {@link CompletionException}, then the cause will be tested.
+	 *
+	 * @return a {@link Uni} with the expected exception thrown by the uni we were testing.
+	 */
+	public static <U extends Throwable> Uni<U> assertThrown(Class<U> expectedException, Uni<?> uni) {
+		return uni.onItemOrFailure().transform( ( s, e )  -> {
+			assertThat( e instanceof CompletionException ? e.getCause() : e ).isInstanceOf( expectedException );
+			return (U) e;
+		});
+	}
+
+	/**
+	 * Asserts that an expected exception and type was thrown
+	 *
+	 * @param throwable
+	 * @param expectedExceptionClass
+	 */
+	public static void assertException(Throwable throwable, Class expectedExceptionClass) {
+		assertThat( throwable ).as( "We were expecting an exception" ).isNotNull();
+		if( expectedExceptionClass != null ) {
+			assertThat( expectedExceptionClass.isInstance( throwable ) );
+		}
+	}
+
+	/**
+	 * Asserts that an exception was thrown and checks that specific text is
+	 * contained within the message
+	 *
+	 * @param throwable
+	 * @param textInMessage text fragments expected in exception's message
+	 */
+	public static void assertMessage(Throwable throwable, String... textInMessage) {
```

Comment:
I think we only use this method and the one after it in one test. I don't think it's worthy to have them as utility at the moment. Maybe in the future.

Reply:
removed